### PR TITLE
Update LIR dumps.

### PR DESCRIPTION
**Pull request by pgavlin**
- Add gtDispRange and gtDispTreeRange for displaying an LIR range
  and the LIR range covered by a tree node and the transitive
  closure of its operands, respectively.
- Add new macros, DISPRANGE and DISPTREERANGE, which wrap calls to
  the above methods s.t. they only execute in debug builds for
  methods specified by {Jit,Ngen}Dump.
- Change all uses of DISPTREE in the backend to whichever of
  DISPNODE and DISPTREERANGE is appropriate for each use.
- Add call argument location information to LIR dumps.
- Remove gtDispLinear*.

### CODE REVIEW

**Comment by pgavlin** [comment]
@BruceForstall PTAL

File: src/jit/jit.h [hunk 1]
```diff
@@ -501,7 +501,9 @@ void JitDump(const char* pcFormat, ...);
 #define JITLOG_THIS(t, x) { (t)->JitLogEE x; }
 #define DBEXEC(flg, expr) if (flg) {expr;}
 #define DISPNODE(t) if (JitTls::GetCompiler()->verbose) JitTls::GetCompiler()->gtDispTree(t, nullptr, nullptr, true);
-#define DISPTREE(x) if (JitTls::GetCompiler()->verbose) JitTls::GetCompiler()->gtDispTree(x)
+#define DISPTREE(t) if (JitTls::GetCompiler()->verbose) JitTls::GetCompiler()->gtDispTree(t);
+#define DISPRANGE(range) if (JitTls::GetCompiler()->verbose) JitTls::GetCompiler()->gtDispRange(range);
+#define DISPTREERANGE(range, t) if (JitTls::GetCompiler()->verbose) JitTls::GetCompiler()->gtDispTreeRange(range, t);
 #define VERBOSE JitTls::GetCompiler()->verbose
```
Comment (BruceForstall) [inline src/jit/jit.h:507]: You'll need empty versions of these below in the !DEBUG section

**Comment by BruceForstall** [comment]
LGTM modulo one comment

File: src/jit/jit.h (hunk 1 above)
Comment (pgavlin) [inline src/jit/jit.h:507]: Argh, I have those locally but forgot to commit them. Will fix.

**Comment by BruceForstall** [comment]
BTW, there are lots of functions at the bottom of compiler.cpp that are useful from the debugger to dump various data structures with minimal arguments needed (so it's easier to use them from the debugger). E.g., dTree(), dTrees(), etc. It would be good if LIR can also be dumped this way. E.g., will dBlocksV() work?

**Comment by pgavlin** [comment]
> E.g., dTree(), dTrees(), etc. It would be good if LIR can also be dumped this way. E.g., will dBlocksV() work?

Pretty much everything except `dTree()` will work without issue. `dTree()` is a bit problematic because it cannot whether or not the tree is LIR and so will display everything as HIR. Not sure how to solve this :(

**Comment by BruceForstall** [comment]
Probably should add a dRange()

**Pull request merged**